lib/improver/wxcode/weather_symbols.py

```diff
@@ -83,16 +82,30 @@ def check_input_cubes(self, cubes):
             for diagnostic, threshold, condition in zip(
                     diagnostics, thresholds, conditions):
 
+                # First we check the diagnostic name and units, performing
+                # a conversion is required and possible.
+                test_condition = (iris.Constraint(name=diagnostic))
+                matched_cube = cubes.extract(test_condition)
+                if not matched_cube:
+                    missing_data.append([diagnostic, threshold, condition])
+                    continue
+                else:
+                    cube_threshold_units = (
+                        matched_cube[0].coord('threshold').units)
+                    threshold.convert_units(cube_threshold_units)
+
+                # Then we check if the required threshold is present in the
+                # cube, and that the thresholding is relative to it correctly.
                 threshold = threshold.points.item()
                 test_condition = (
-                    Constraint(
-                        name=diagnostic,
-                        coord_values={'threshold': threshold},
+                    iris.Constraint(
+                        threshold=lambda cell:
+                            threshold*0.99 < cell < threshold*1.01,
```

Comment:
You use these "magic numbers", in two places in this class. It might be nice to put them in one place (like in the __init__) so that if we do need to change them it's easier. What do you think?

Reply:
Done.